testsuite/classlibrary/TestEnvironmentRedirect.sc

```diff
@@ -0,0 +1,28 @@
+TestEnvironmentRedirect : UnitTest {
+
+	test_dispatch {
+		var e = EnvironmentRedirect();
+		var notified = false;
+		e.dispatch = { notified = true };
+
+		e[\test] = 1;
+		this.assert(notified, "should run dispatch when a new value is put");
+
+		notified = false;
+		e[\test] = 2;
+		this.assert(notified, "should run dispatch when an existing value is modified");
+
+		notified = false;
+		e[\test] = nil;
+		this.assert(notified, "should run dispatch when an existing value is set to nil");
+		e[\test] = 2;
+
+		notified = false;
+		e.removeAt(\test);
+		this.assert(notified, "should run dispatch when an existing value is removed");
+
```

Comment:
maybe add an additional test that `\test` does indeed not trigger a callback after its deletion? I am sure it won't but that is what we are testing, right?

We should also check the arguments passed to the dispatcher?

Reply:
ok, done in https://github.com/supercollider/supercollider/pull/5923/commits/730df88f55b1a55e8707d34afcbe9f5bb46219ba